realm/realm-library/src/main/cpp/io_realm_internal_CollectionChangeSet.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "io_realm_internal_CollectionChangeSet.h"
+
+#include <collection_notifications.hpp>
+
+#include "util.hpp"
+
+using namespace realm;
+
+static void finalize_changeset(jlong ptr);
+static jlongArray index_set_to_jlong_array(JNIEnv* env, const IndexSet& index_set);
+static jlongArray index_set_to_indices_array(JNIEnv* env, const IndexSet& index_set);
+
+static void finalize_changeset(jlong ptr)
+{
+    TR_ENTER_PTR(ptr);
+    delete reinterpret_cast<CollectionChangeSet*>(ptr);
+}
+
+static jlongArray index_set_to_jlong_array(JNIEnv* env, const IndexSet& index_set)
+{
+    if (index_set.empty()) {
+        return env->NewLongArray(0);
+    }
+
+    std::vector<jlong> ranges_vector;
+    for (auto& changes : index_set) {
+        ranges_vector.push_back(changes.first);
+        ranges_vector.push_back(changes.second - changes.first);
+    }
+
+    if (ranges_vector.size() > io_realm_internal_CollectionChangeSet_MAX_ARRAY_LENGTH) {
+        ThrowException(env, IllegalState,
```

Comment:
If `ranges_vector.size()` and `io_realm_internal_CollectionChangeSet_MAX_ARRAY_LENGTH` are included in the error message, user might have a better chance to find the bug.